bin/cron/sync_dropbox

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env ruby
+
+# This syncs content changes from the shared dropbox folder to our github repo
+# on the staging machine. This is required for contents changes from dropbox to
+# be deployed.
+
+SCRIPT_START = Time.now
+
+require_relative "../../lib/cdo/only_one"
+require_relative "../../deployment"
+require "cdo/slack"
+require "open3"
+exit unless only_one_running?(__FILE__)
+
+FOLDERS = {
+  "advocacy.code.org" => "sites.v3/advocacy.code.org",
+  "applab-docs-1" => "sites.v3/code.org/public/applab/docs1",
+  "applab-docs" => "sites.v3/code.org/public/applab/docs",
+  "code.org" => "sites.v3/code.org",
+  "csedweek.org" => "sites.v3/csedweek.org",
+  "curriculum-6-8" => "sites/virtual/curriculum-6-8",
+  "curriculum-algebra" => "sites/virtual/curriculum-algebra",
+  "curriculum-course1" => "sites/virtual/curriculum-course1",
+  "curriculum-course2" => "sites/virtual/curriculum-course2",
+  "curriculum-course3" => "sites/virtual/curriculum-course3",
+  "curriculum-course4" => "sites/virtual/curriculum-course4",
+  "curriculum-csp" => "sites/virtual/curriculum-csp",
+  "curriculum-docs" => "sites/virtual/curriculum-docs",
+  "curriculum-misc" => "sites/virtual/curriculum-misc",
+  "curriculum-science" => "sites/virtual/curriculum-science",
+  "curriculum-unplugged" => "sites/virtual/curriculum-unplugged",
+  "emails" => "emails",
+  "hourofcode.com" => "sites.v3/hourofcode.com"
+}.freeze
+
+INTERVAL_SECONDS = 5
+TOTAL_SECONDS = 60 - INTERVAL_SECONDS
+error_message = ""
+alert_slack = true
+while (Time.now - SCRIPT_START) < TOTAL_SECONDS
+  attempt_start = Time.now
+  FOLDERS.each do |key, value|
+    # 'unison' will sync the two folders. It will return true on success and false on failure.
+    # For full details, see unison's man page. Docs are also here: https://www.cis.upenn.edu/~bcpierce/unison/download/releases/stable/unison-manual.html
+    command = "unison /home/ubuntu/Dropbox/Shared/#{key} /home/ubuntu/staging/pegasus/#{value} -silent -ignore \"Name .dropbox\" -auto -perms 0 -dontchmod"
+    if key == "code.org"
+      # The applab-docs and applab-docs-1 folders on dropbox are synced separately even though
+      # they are sub-paths of the code.org folder on git.
+      command << " -ignore \"Path */public/applab/docs\" -ignore \"Path */public/applab/docs1\""
+    end
+    stdout, _, _ = Open3.capture3(command)
+    if stdout != ""
+      error_message = error_message + "Dropbox #{key}. staging #{value}. " + stdout + "\n"
+    end
+  end
+  current_time = Time.now
+  if alert_slack && error_message != ""
+    Slack.message("@here Error syncing Dropbox and staging\n" + error_message, {channel: "sync-dropbox-staging"})
```

Comment:
Could we publish this to `#infra-staging` instead?  Assuming the frequency of errors is low, the Developer of the Day could notify content editors.

Reply:
We talked about this and decided to leave this as a TODO (to follow up on in a timely fashion), since it requires a bit of logic to dedupe the slack notification (probably via a local file.)